#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/obstacles/protos/OilBarrel.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/toys/protos/RubberDuck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/balls/protos/SoccerBall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/drinks/protos/Can.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/fruits/protos/Apple.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5711399008772076 -0.004055857391807745 0.8208427155349499 3.1532556352982053
  position 1.6490209744814184 0.07711590694356822 3.6386694850825174
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
  floorAppearance Parquetry {
    type "light strip"
  }
}
Apple {
  translation 0.52 0.34 0.05
  rotation -1 0 0 -5.307179586466759e-06
}
Can {
  translation -0.25 -3.0979e-17 0.0609627
  rotation 0.12108010303956522 0.10152078896415774 -0.9874376628709395 2.240919065423009e-16
}
SoccerBall {
  translation 0.53 0.62 0.106058
  radius 0.114
}
RubberDuck {
  translation -0.15 -0.59 0.020438
  rotation 0.14873919671761285 -0.0697988301243212 0.9864100438829054 5.42853403859421e-16
}
OilBarrel {
  translation -0.62 -0.57 0.1
  rotation 0 1 0 0
  height 0.2
  radius 0.086
}
DEF Robot Robot {
  translation 0.95 0.91 0
  rotation 0 0 1 -3.141592653589793
  children [
    Transform {
      translation 0 0 0.0415
      rotation 0 0 1 1.570796
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 985.0748717753905
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
      ]
      endPoint Solid {
        translation 0 0.04500000000000027 0.025
        rotation -0.5933273107765037 0.5933269366057067 0.5439906695751406 4.1380275923824525
        children [
          DEF WHEEL Transform {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left_wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 3707.2452907405473
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
      ]
      endPoint Solid {
        translation 0 -0.04499999999999898 0.025
        rotation -0.14348597015557032 0.14348596286251467 -0.9791953609111574 1.5918189725427363
        children [
          USE WHEEL
        ]
        name "right_wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/samples/devices/worlds/textures/mybot.png"
          ]
        }
        roughness 0.5
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.015 0.038 -0.041
            0.015 0.023 -0.041
            0 0.023 -0.0455
            -0.015 0.023 -0.041
            -0.015 0.038 -0.041
            0 0.038 -0.0455
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0.5 0
            1 0
            1 1
            0.5 1
            0 1
          ]
        }
        coordIndex [
          0, 1, 2, 5, -1, 5, 2, 3, 4, -1
        ]
        texCoordIndex [
          5, 0, 1, 4, -1, 4, 1, 2, 3, -1
        ]
        creaseAngle 0.9
      }
    }
    TouchSensor {
      translation 0.04580618039996165 -0.00012127100056891216 0.019862109379781955
      rotation 0.9999999999999976 -4.963782217688971e-08 -5.193535659053206e-08 -3.141592308374597
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.6980390000000001
            metalness 0
          }
          geometry DEF BUMPER Box {
            size 0.02 0.09 0.02
          }
        }
      ]
      name "bumper"
      boundingObject USE BUMPER
      physics Physics {
        density -1
        mass 0.01
      }
      lookupTable [
        0 0 0
        0.01 1 0
      ]
    }
    Camera {
      translation -0.0085 0.04 0.07
      rotation -7.8088e-07 1.36929e-06 -1 -1.5707953071795862
      fieldOfView 0.25
      width 250
      height 200
      recognition Recognition {
        maxObjects 1
        frameColor 1 1 0
        frameThickness 3
      }
    }
  ]
  name "Robot"
  boundingObject Transform {
    translation 0 0 0.0415
    rotation 0 0 1 1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "Bump"
}
